src/raft.rs

```diff
@@ -235,9 +235,26 @@ pub fn vote_resp_msg_type(t: MessageType) -> MessageType {
 impl<T: Storage> Raft<T> {
     /// Creates a new raft for use on the node.
     #[allow(clippy::new_ret_no_self)]
-    pub fn new(c: &Config, store: T, logger: &Logger) -> Result<Raft<T>> {
+    pub fn new<'a>(c: &Config, store: T, logger: impl Into<Option<&'a Logger>>) -> Result<Raft<T>> {
         c.validate()?;
-        let logger = logger.new(o!("raft_id" => c.id));
+
+        let logger = logger.into();
+        #[cfg(feature = "default-logger")]
```

Comment:
How about providing a new function `with_default_logger` instead?

Reply:
Yes! I think having `new(config, store).with_logger(x)` or `new(config, store).with_default_logger()` is good :)

Reply:
Oh, I mean `with_default_logger` works like `with_capacity` as a constructor.

Reply:
I don't think this is a better solution but since it doesn't really matter (We'll move off slog later anyways) letr's do it your way and get this merged.